metaflow/plugins/aws/batch/batch_client.py

```diff
@@ -14,42 +15,118 @@
 
 from metaflow.exception import MetaflowException
 from metaflow.metaflow_config import AWS_SANDBOX_ENABLED
+from metaflow.metaflow_config import BATCH_DESCRIBE_JOBS_THROTTLE_DELTA as DELTA
+from metaflow.metaflow_config import BATCH_DESCRIBE_JOBS_THROTTLE_TRIES as TRIES
+
+
+class Throttle(object):
+    def __init__(self, delta_in_secs=DELTA, num_tries=TRIES):
+        self._now = time.time()
+        self.delta_in_secs = int(delta_in_secs)  # Cast if from env variable
+        self.num_tries = int(num_tries)  # Cast if from env variable
+        self._reset()
+
+    def _reset(self):
+        self._tries_left = self.num_tries
+
+    def __call__(self, func):
+        def wrapped(*args, **kwargs):
+            # Don't poll incessantly
+            wait = self.delta_in_secs - (time.time() - self._now)
+            if wait > 0:
+                time.sleep(wait)
+                self._now = time.time()
+            # Throttle til failure
+            while "throttling":
+                try:
+                    value = func(*args, **kwargs)
+                except TriableException as ex:
+                    self._tries_left -= 1
+                    if self._tries_left == 0:
+                        raise ex.ex
+                    backoff = (self.delta_in_secs * 1.2) ** (
+                        self.num_tries - self._tries_left
+                    )
+                    jitter = random.randint(0, 3 * self.delta_in_secs)
+                    time.sleep(backoff + jitter)
+                else:
+                    self._reset()
+                    return value
+
+        return wrapped
+
+
+@contextmanager
+def manage_client_error(batch_client):
```

Comment:
There gotta be a more intuitive name, maybe `convert_retryable_errors(...)` or something? Otherwise when I see this in code, it is unclear what "manage" means

Reply:
this has now been converted into an `exception_handler` in the new refactor and so is clearer (I hope)